### DISCUSSION TITLE
OpenID Connect middleware

### DISCUSSION THREAD

**Discussion by simongottschlag** [go-goyave/goyave#164] [Ideas]
Hi!

Cool project! 😊

I've been building a generic OIDC middleware for go http frameworks: https://github.com/XenitAB/go-oidc-middleware

It's built to easily integrate with OpenID Providers like Azure AD, Okta, Auth0, Cognito etc. Handling discovery and validation of the JWTs.

Do you think it would provide any benefit to goyave if I created a middleware for it as well?

**Comment by System-Glitch** [comment]
Hello and thank you for your interest in Goyave!

Your middleware looks very useful! I think it could be awesome to have stronger authentication mechanics for Goyave.

If you need any help or guidance about how you could implement it, don't hesitate to ask me.

**Reply by simongottschlag** [reply]
Hi @System-Glitch!

Cool! I'm doing this on my free time, but will take a look at it when I have inspiration and get back to you when I'm getting close :+1:

**Comment by System-Glitch** [comment]
A very old conversation I know, but I recently released an Auth0 authenticator for the framework: https://github.com/go-goyave/goyave-auth0
